src/traits.rs

```diff
@@ -21,8 +21,28 @@ impl<N> NodeIdT for N where N: Eq + Ord + Clone + Debug + Hash + Send + Sync {}
 pub trait Message: Debug + Send + Sync {}
 impl<M> Message for M where M: Debug + Send + Sync {}
 
-/// Result of one step of the local state machine of a distributed algorithm. Such a result should
-/// be used and never discarded by the client of the algorithm.
+/// Single algorithm step result.
+///
+/// Each time input (typically in the form of user input or incoming network messages) is provided
+/// to an instance of an algorithm, a `Step` is produced as a result, potentially containing
+/// output values, fault log and network messages.
```

Comment:
a fault log,

Reply:
I was not sure what kind of entities the fault log contains and whether it's technically a single log or multiple logs. I decided to cheat by leaving it ambivalent ("[fault log and network] messages" vs "[fault log] and [network messages]").

Changing it now, @afck will correct me if it's wrong =).

Reply:
Our `FaultLog` type contains several `Fault`s, each of which blames a `FaultKind` on a `NodeId`.

I'm fine with "a fault log". :+1: